src/pages/Profile.js

```diff
@@ -58,6 +59,107 @@ const styles = {
+    setNihStatus(status)
+  }
+
+  const linkNihAccount = async nihToken => {
+    setLinking(true)
+    const status = await Ajax().User.linkNihAccount(nihToken)
+    setNihStatus(status)
+    setLinking(false)
+  }
+
+
+  /*
+   * Hooks
+   */
+  const [{ linkedNihUsername, linkExpireTime, datasetPermissions }, setNihStatus] = useState({})
+  const [linking, setLinking] = useState(false)
+
+  Utils.useOnMount(() => {
+    if (nihToken) {
+      // Clear the query string, but use replace so the back button doesn't take the user back to the token
+      Nav.history.replace({ search: '' })
+      linkNihAccount(nihToken)
+    } else {
+      loadNihStatus()
+    }
+  })
+
+
+  /*
+   * Render helpers
+   */
+  const makeAccountLinkLink = label => {
+    const nihRedirectUrl = `${window.location.protocol}//${window.location.host}/#profile?nih-username-token={token}`
+
+    return link({
+      href: `${getConfig().shibbolethUrlRoot}/link-nih-account?redirect-url=${encodeURIComponent(nihRedirectUrl)}`,
+      style: { display: 'flex', alignItems: 'center' },
+      target: '_blank'
+    }, [
+      label,
+      icon('pop-out', { size: 12, style: { marginLeft: '0.5rem' } })
+    ])
+  }
+
+  const makeDatasetAuthStatus = ({ name, authorized }) => {
+    return div({ key: `nih-auth-status-${name}`, style: { display: 'flex' } }, [
+      div({ style: { flex: 1 } }, [`${name} Authorization`]),
+      div({ style: { flex: 2 } }, [
+        authorized ? 'Authorized' : 'Not Authorized',
+        !authorized && h(InfoBox, { style: { marginLeft: '0.5rem' } }, [
+          'Your account was linked, but you are not authorized to view this controlled dataset. Please go ',
+          link({
+            href: 'https://dbgap.ncbi.nlm.nih.gov/aa/wga.cgi?page=login',
+            target: '_blank'
+          }, [
+            'here',
+            icon('pop-out', { size: 12 })
+          ]),
+          ' to check your credentials.'
+        ])
+      ])
+    ])
+  }
+
+
+  /*
+   * Render
+   */
+  return h(Fragment, [
+    div({ style: styles.form.title }, [
+      'NIH Account',
+      h(InfoBox, { style: { marginLeft: '0.5rem' } }, [
+        'Linking with eRA Commons will allow FireCloud to automatically determine if you can access controlled datasets hosted in FireCloud (ex. TCGA) based on your valid dbGaP applications.'
+      ])
+    ]),
+    linking && div([spinner(), 'Linking NIH account...']),
+    !!linkedNihUsername && div({ style: { display: 'flex', flexDirection: 'column', width: '35rem' } }, [
+      div({ style: { display: 'flex' } }, [
+        div({ style: { flex: 1 } }, ['Username:']),
+        div({ style: { flex: 2 } }, [linkedNihUsername])
+      ]),
+      div({ style: { display: 'flex' } }, [
+        div({ style: { flex: 1 } }, ['Link Expiration:']),
+        div({ style: { flex: 2 } }, [
+          Utils.makeCompleteDate(linkExpireTime * 1000),
+          makeAccountLinkLink('Log-In to NIH to re-link your account')
```

Comment:
Should we still show this if their link hasn't expired yet?

Reply:
This is what FireCloud does. Note that it says "re-link". I'm guessing there's a good reason why someone might need to do this.

Reply:
Just as likely a historical artifact.

One thing that FireCloud does not have is the ability to unlink—Example, I no longer need the link but I see the "expired" banner perpetually. A worthy follow-on story, in my opinion.

Reply:
Agree. I didn't notice any existing unlink API, otherwise I'd add it right now.